### PR TITLE
DhtId to NameType

### PR DESCRIPTION
**Pull request by krishnaIndia**

### CODE REVIEW

File: src/name_type.rs [hunk 1]
```diff
@@ -92,11 +100,53 @@ impl fmt::Debug for NameType {
...
+  #[inline]
+  fn ge(&self, other : &NameType) -> bool {
+    PartialOrd::ge(&&self.0[..], &&other.0[..])
+  }
 }
 
+// FIXME(Ben): please fix me
+impl hash::Hash for NameType {
+  fn hash<H: hash::Hasher>(&self, state: &mut H) {
+    state.write(&self.0[..])
```
Comment (ustulation) [inline src/name_type.rs]: The example shows something like this:
```
impl Hash for Person {
    fn hash<H: Hasher>(&self, state: &mut H) {
        self.id.hash(state);
        self.phone.hash(state);
    }
}
```
Is this the same ?

Comment (krishnaIndia) [inline src/name_type.rs]: Am not sure either. But I think, Hash trait is not implemented for [T;64], from the documentation (https://doc.rust-lang.org/std/hash/trait.Hash.html). So not sure, whether hash() can be called directly as in the example.

**Comment by benjaminbollen** [comment]
Thanks Krishna !

**Comment by krishnaIndia** [comment]
Cheers Ben :)

**Pull request merged**